### PR TITLE
test: limit-count add more test cases to cover the case where the upstream is a domain name

### LINKED ISSUE #3108: bug: limit-count plugin does not take effect
### Issue description

When the `limit-count` plugin is enabled on `service` and `upstream` is the domain name. This will cause the `local key = (ctx.var[conf.key] or "") .. ctx.conf_type .. ctx.conf_version` generated in the limit-count plugin to be inconsistent. When `ctx.conf_version` is deleted (`local key = (ctx.var[conf.key] or "") .. ctx.conf_type`), the plug-in function returns to normal.

Steps to reproduce, will be added later.

### Environment

* apisix version (cmd: `apisix version`): 2.1
* OS: (cmd: `uname -a`)：centos7
* OpenResty / Nginx version: (cmd: `nginx -V` or `openresty -V`): openresty/1.19.3.1

### Minimal test code / Steps to reproduce the issue
1.
2.
3.

### PR DESCRIPTION
**Pull request by Firstsawyou**
### What this PR does / why we need it:

close #3108

 `limit-count` plugin add more test cases to cover the case where the upstream is a domain name.

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: t/plugin/limit-count.t [hunk 1]
```diff
@@ -1175,3 +1175,95 @@ passed
...
+--- request
+GET /t
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 38: add route and bind service
```
Comment (spacewander) [inline t/plugin/limit-count.t:1220]: I think we should merge TEST 38 into TEST 37, so that it is clear that they are both for TEST 39.

Comment (Firstsawyou) [reply t/plugin/limit-count.t]: updated.

File: t/plugin/limit-count.t [hunk 2]
```diff
@@ -1175,3 +1175,73 @@ passed
 [200, 200, 503, 503]
 --- no_error_log
 [error]
+
+
+
+=== TEST 37: add route and upstream is the domain name
```
Comment (spacewander) [inline t/plugin/limit-count.t:1181]: The #3108 is limit-count on the `service`, not a route?

**Review by spacewander — CHANGES_REQUESTED** [review]

File: t/plugin/limit-count.t (hunk 2 above)
Comment (Firstsawyou) [reply t/plugin/limit-count.t:1181]: The main test is the case where the upstream is a domain name, which is also feasible.
But I am adding some tests that limit the service.

File: t/plugin/limit-count.t [hunk 3]
```diff
@@ -1175,3 +1175,170 @@ passed
...
+--- request
+GET /t
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 38: add route and upstream is the domain name
```
Comment (spacewander) [inline t/plugin/limit-count.t:1214]: TEST 38 should be merged into TEST 37.

**Review by spacewander — CHANGES_REQUESTED** [review]

File: t/plugin/limit-count.t (hunk 3 above)
Comment (Firstsawyou) [reply t/plugin/limit-count.t]: updated.

File: t/plugin/limit-count.t [hunk 4]
```diff
@@ -1178,7 +1178,7 @@ passed
 
 
 
-=== TEST 37: plugin is bound to the service
+=== TEST 37: plugin is bound to the service and upstream is the domain name
```
Comment (spacewander) [inline t/plugin/limit-count.t:1181]: I think we can also add route and bind service_id in this one.

Comment (Firstsawyou) [reply t/plugin/limit-count.t]: updated.

**Review by spacewander — APPROVED** [review]

**Review by gxthrj — APPROVED** [review]

**Pull request merged**